### ISSUE TITLE
The algorithm predicts all black images.

### ISSUE THREAD

**Issue by maqorbani** [yueAUW/neural-daylighting#2]
Hey Yue Liu!
I've been trying to run the algorithm with the 16 bit data you provided, but I had some issues with code that I had hoped that you could help me with.
First of all, I get this error from the code while running the plotLoss function "plotLoss(history)" in predictLumMaps.py:

`
Epoch 00030: ReduceLROnPlateau reducing learning rate to 1e-13.

Epoch 00030: val_loss did not improve from inf
65/65 - 24s - loss: inf - mean_squared_error: nan - mean_absolute_error: nan - relative_err: inf - mse_log: inf - val_loss: inf - val_mean_squared_error: nan - val_mean_absolute_error: nan - val_relative_err: inf - val_mse_log: 0.0925 - lr: 1.8626e-13
batch size reduce to: 1
loss is: inf
target loss is: 1e-05
min batch size is: 1
Saved trained model to disk
dict_keys(['loss', 'mean_squared_error', 'mean_absolute_error', 'relative_err', 'mse_log', 'val_loss', 'val_mean_squared_error', 'val_mean_absolute_error', 'val_relative_err', 'val_mse_log', 'lr'])
Traceback (most recent call last):
  File "e:\University-Master-Building science\New Proposal\neural-daylighting\main.py", line 57, in <module>
    main()
  File "e:\University-Master-Building science\New Proposal\neural-daylighting\main.py", line 52, in main
    train_model(**kwargs)
  File "e:\University-Master-Building science\New Proposal\neural-daylighting\deep_light\predictLumMaps.py", line 1976, in train_model
    learning_rate=LR)
  File "e:\University-Master-Building science\New Proposal\neural-daylighting\deep_light\predictLumMaps.py", line 1817, in train
    plotLoss(history)
  File "e:\University-Master-Building science\New Proposal\neural-daylighting\deep_light\predictLumMaps.py", line 1283, in plotLoss
    plt.ylim([0, history.history['loss'][0]])
  File "C:\Users\qorba\anaconda3\lib\site-packages\matplotlib\pyplot.py", line 1471, in ylim
    ret = ax.set_ylim(*args, **kwargs)
  File "C:\Users\qorba\anaconda3\lib\site-packages\matplotlib\axes\_base.py", line 3623, in set_ylim
    top = self._validate_converted_limits(top, self.convert_yunits)
  File "C:\Users\qorba\anaconda3\lib\site-packages\matplotlib\axes\_base.py", line 3158, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
`

Then I commented out the plotLoss function so that code goes ahead as the function did just plotting and it would not break the algorithm.
And when the algorithm produced the results data, this is the HDR results: (left the target image and right the prediction)

![image](https://user-images.githubusercontent.com/47574645/91016002-52feaf80-e601-11ea-82c2-a76a0a0b7550.png)

What can I do about these issues? Thanks in forward.

**Comment by alexcolburn** [comment]
Hi  Mohammad,

Thanks for trying out the code & reporting an issue!

It looks like the training failed - evident from the training loss with inf and NaN values.

In my experience there are 2 main culprits here.  1) tensorflow versions, and 2) 16bit data conversions.
There might also be various platform dependencies - I think this was only tested on a Windows 10 machines.

Can you create a fresh conda environment and make sure that you install tensorflow-2.1.0?   Could you also try the FLOAT32 version of the dataset?

Thanks

**Comment by maqorbani** [comment]
Thank you very much for responding, I'll try these and I'll share the results.

**Comment by maqorbani** [comment]
The issue is solved, thank you very much!

I would suggest adding a os.path.exist in the main.py for checking ALL_DATA_FP16, 32, and 64 so that the algorithm would automatically find the data directory as opposed to hardcoding ALL_DATA_FP16.

Thanks

**Comment by alexcolburn** [comment]
Great!

What was the bug?  was it the tensorflow version, data format? or something else?
Feel free to make bug fixes!

-Alex

**Comment by maqorbani** [comment]
Thank you, Alex.

First, I made all of the changes you mentioned as culprits, namely: TensorFlow version 2.1.0 and a fresh conda environment and of course 32-bit data, simultaneously. And it worked well.
Then, I ran the model again with the TensorFlow version 2.1.0 and the same new environment with 16-bit data. And the algorithm broke then. So I infer that the 16-bit data was making the issue. Although I think this needs a little bit more experimenting. I have saved all of my runtime logs and I can share if they help.

If I were to fix bugs should I request on the master branch or make a new one?
